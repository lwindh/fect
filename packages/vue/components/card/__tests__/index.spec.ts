import { mount } from '@vue/test-utils'
import Card from '..'

const Wrapper = {
  component: {
    [Card.name]: Card,
  },
  data() {
    return {
      hoverable: false,
      shadow: false,
    }
  },
  template: `
   <div class="container">
    <fe-card>Normalize Card</fe-card>
    <fe-card :hoverable="hoverable">Hoverable Card</fe-card>
    <fe-card :shadow="shadow">Shadow Card</fe-card>
    <fe-card :hoverable="hoverable" :shadow="shadow">Hoverable and Shadow Card</fe-card>
   </div>
  `,
}

describe('Card', () => {
  it('should be render as a element', () => {
    const wrapper = mount(Card)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should support render different state component', async () => {
    const wrapper = mount(Wrapper)
    await wrapper.setData({ hoverable: true, shadow: true })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
