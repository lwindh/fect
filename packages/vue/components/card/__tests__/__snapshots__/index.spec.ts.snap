// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should be render as a element 1`] = `
<div class="fect-card ">
  <div class="fect-card__content">
    <!---->
  </div>
</div>
`;

exports[`Card should support render different state component 1`] = `
<div class="container">
  <fe-card>Normalize Card</fe-card>
  <fe-card hoverable="true">Hoverable Card</fe-card>
  <fe-card shadow="true">Shadow Card</fe-card>
  <fe-card hoverable="true" shadow="true">Hoverable and Shadow Card</fe-card>
</div>
`;
